src/main_context.rs

```diff
@@ -96,7 +96,7 @@ impl MainContext {
     }
 }
 
-#[cfg_attr(feature = "cargo-clippy", allow(transmute_ptr_to_ref))]
+#[cfg_attr(feature = "cargo-clippy", allow(clippy::transmute_ptr_to_ref))]
 unsafe extern "C" fn trampoline<F: FnOnce() + 'static>(func: gpointer) -> gboolean {
     let func: &mut Option<F> = transmute(func);
```

Comment:
This could simply be a cast: `let func: &mut Option<F> = &*(func as *mut Option<F>)`